src/binance/bnbEngine.ts

```diff
@@ -146,14 +146,12 @@ export class BinanceEngine extends CurrencyEngine<
         }
       }
     } catch (e: any) {
-      // fetching of account balances for uninitiated accounts returns 404 (throws error)
-      if (
-        this.tokenCheckTransactionsStatus.BNB === 1 &&
-        this.transactionList.BNB.length === 0
-      ) {
+      if (e.message?.includes('error code 404') === true) {
+        // fetching of account balances for uninitiated accounts returns 404 (throws error)
         this.updateBalance('BNB', '0')
+      } else {
+        this.error(`Error checking BNB address balance ${e.message}`)
```

Comment:
If `e` happens to be `null`, this error handling will crash. You could do `asObject({ message: asString })` to check the error before accessing the message, or you could do `/error code 404/.test(e?.message)`, which doesn't rely on the message existing or even being a string.

Reply:
For the line `this.error('Error checking BNB address balance ${e.message}')`, it to `this.error('Error checking BNB address balance ${String(e)}')`. That will turn the error object into a string more safely.